Applications/Spire/Source/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.cpp

```diff
@@ -0,0 +1,46 @@
+#include "Spire/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.hpp"
+
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+OrderImbalanceIndicatorTableModel::OrderImbalanceIndicatorTableModel(
+    TimeInterval interval,
+    std::shared_ptr<OrderImbalanceIndicatorModel> source)
+    : m_source(std::move(source)) {
```

Comment:
Why is the source stored in the class? It isn't used in other member functions except in the constructor.

Reply:
It keeps a pointer to it so the subscription won't be cancelled if whatever else is holding a pointer to the OrderImbalanceIndicatorModel is destroyed.